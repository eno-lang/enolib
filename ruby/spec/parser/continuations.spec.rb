input = <<~DOC.strip
Field: Value
\\ Spaced line continuation
| Direct line continuation

Fieldset:
Entry = Value
\\ Spaced line continuation
| Direct line continuation

List:
- Value
\\ Spaced line continuation
| Direct line continuation

Empty field:
\\ Spaced line continuation
| Direct line continuation

Fieldset with empty entry:
Empty entry =
\\ Spaced line continuation
| Direct line continuation

List with empty item:
-
\\ Spaced line continuation
| Direct line continuation
DOC

context = {
  indexing: 1,
  input: input,
  messages: Enolib::Messages::En,
  reporter: Enolib::TextReporter
}

describe Enolib::Parser do
  describe 'Continuations' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
