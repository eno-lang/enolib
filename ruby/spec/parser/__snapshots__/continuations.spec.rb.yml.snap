---
"./spec/parser/continuations.spec.rb[1:1:1]": &1
  :elements:
  - :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 12
      :element_operator:
      - 5
      - 6
      :key:
      - 0
      - 5
      :value:
      - 7
      - 12
    :key: Field
    :type: :field
    :value: Value
    :parent: *1
    :continuations:
    - :index: 13
      :line: 1
      :ranges:
        :line:
        - 13
        - 39
        :spaced_line_continuation_operator:
        - 13
        - 14
        :value:
        - 15
        - 39
      :spaced: true
      :type: :continuation
      :value: Spaced line continuation
    - :index: 40
      :line: 2
      :ranges:
        :line:
        - 40
        - 66
        :direct_line_continuation_operator:
        - 40
        - 41
        :value:
        - 42
        - 66
      :type: :continuation
      :value: Direct line continuation
  - &2
    :index: 68
    :line: 4
    :ranges:
      :line:
      - 68
      - 77
      :element_operator:
      - 76
      - 77
      :key:
      - 68
      - 76
    :key: Fieldset
    :type: :fieldset
    :parent: *1
    :entries:
    - :index: 78
      :line: 5
      :ranges:
        :line:
        - 78
        - 91
        :key:
        - 78
        - 83
        :entry_operator:
        - 84
        - 85
        :value:
        - 86
        - 91
      :type: :fieldset_entry
      :key: Entry
      :value: Value
      :parent: *2
      :continuations:
      - :index: 92
        :line: 6
        :ranges:
          :line:
          - 92
          - 118
          :spaced_line_continuation_operator:
          - 92
          - 93
          :value:
          - 94
          - 118
        :spaced: true
        :type: :continuation
        :value: Spaced line continuation
      - :index: 119
        :line: 7
        :ranges:
          :line:
          - 119
          - 145
          :direct_line_continuation_operator:
          - 119
          - 120
          :value:
          - 121
          - 145
        :type: :continuation
        :value: Direct line continuation
  - &3
    :index: 147
    :line: 9
    :ranges:
      :line:
      - 147
      - 152
      :element_operator:
      - 151
      - 152
      :key:
      - 147
      - 151
    :key: List
    :type: :list
    :parent: *1
    :items:
    - :index: 153
      :line: 10
      :ranges:
        :line:
        - 153
        - 160
        :item_operator:
        - 153
        - 154
        :value:
        - 155
        - 160
      :type: :list_item
      :value: Value
      :parent: *3
      :continuations:
      - :index: 161
        :line: 11
        :ranges:
          :line:
          - 161
          - 187
          :spaced_line_continuation_operator:
          - 161
          - 162
          :value:
          - 163
          - 187
        :spaced: true
        :type: :continuation
        :value: Spaced line continuation
      - :index: 188
        :line: 12
        :ranges:
          :line:
          - 188
          - 214
          :direct_line_continuation_operator:
          - 188
          - 189
          :value:
          - 190
          - 214
        :type: :continuation
        :value: Direct line continuation
  - :index: 216
    :line: 14
    :ranges:
      :line:
      - 216
      - 228
      :element_operator:
      - 227
      - 228
      :key:
      - 216
      - 227
    :key: Empty field
    :type: :field
    :parent: *1
    :continuations:
    - :index: 229
      :line: 15
      :ranges:
        :line:
        - 229
        - 255
        :spaced_line_continuation_operator:
        - 229
        - 230
        :value:
        - 231
        - 255
      :spaced: true
      :type: :continuation
      :value: Spaced line continuation
    - :index: 256
      :line: 16
      :ranges:
        :line:
        - 256
        - 282
        :direct_line_continuation_operator:
        - 256
        - 257
        :value:
        - 258
        - 282
      :type: :continuation
      :value: Direct line continuation
  - &4
    :index: 284
    :line: 18
    :ranges:
      :line:
      - 284
      - 310
      :element_operator:
      - 309
      - 310
      :key:
      - 284
      - 309
    :key: Fieldset with empty entry
    :type: :fieldset
    :parent: *1
    :entries:
    - :index: 311
      :line: 19
      :ranges:
        :line:
        - 311
        - 324
        :key:
        - 311
        - 322
        :entry_operator:
        - 323
        - 324
      :type: :fieldset_entry
      :key: Empty entry
      :parent: *4
      :continuations:
      - :index: 325
        :line: 20
        :ranges:
          :line:
          - 325
          - 351
          :spaced_line_continuation_operator:
          - 325
          - 326
          :value:
          - 327
          - 351
        :spaced: true
        :type: :continuation
        :value: Spaced line continuation
      - :index: 352
        :line: 21
        :ranges:
          :line:
          - 352
          - 378
          :direct_line_continuation_operator:
          - 352
          - 353
          :value:
          - 354
          - 378
        :type: :continuation
        :value: Direct line continuation
  - &5
    :index: 380
    :line: 23
    :ranges:
      :line:
      - 380
      - 401
      :element_operator:
      - 400
      - 401
      :key:
      - 380
      - 400
    :key: List with empty item
    :type: :list
    :parent: *1
    :items:
    - :index: 402
      :line: 24
      :ranges:
        :line:
        - 402
        - 403
        :item_operator:
        - 402
        - 403
      :type: :list_item
      :parent: *5
      :continuations:
      - :index: 404
        :line: 25
        :ranges:
          :line:
          - 404
          - 430
          :spaced_line_continuation_operator:
          - 404
          - 405
          :value:
          - 406
          - 430
        :spaced: true
        :type: :continuation
        :value: Spaced line continuation
      - :index: 431
        :line: 26
        :ranges:
          :line:
          - 431
          - 457
          :direct_line_continuation_operator:
          - 431
          - 432
          :value:
          - 433
          - 457
        :type: :continuation
        :value: Direct line continuation
  :type: :document
